### PR TITLE
Odometry Docs

### PR DESCRIPTION
**Pull request by theol0403**
### Description of the Change

This PR adds an odometry walkthrough tutorial, as well as various documentation improvements.

The odometry walkthrough is not yet completed, but I am submitting what I have so far so that the review process can get started. I will have time to finish the tutorial tomorrow night.

**Changes**:

- Remove unnecessary period from title - it was causing double period in footer.
- Remove all `using namespace okapi` from examples - it did not add value to the examples and users are expected to include the using namespace from `main.h` anyways.
- Add note about following motion profile paths backwards
- Format motion profile examples
- Add odometry walkthrough
- Improve main page
- Remove `LogLevel::warn` link in logger tutorial to encourage clicking on the link to `LogLevel`.

### Motivation

This will help users understand how to use odometry.

### CODE REVIEW

**Comment by Octogonapus** [comment]
> Remove all using namespace okapi from examples - it did not add value to the examples and users are expected to include the using namespace from main.h anyways.

I kinda disagree with this. We see users in Discord not know to include the namespace, so I put it in the examples to make it clear that it uses the namespace.

File: docs/tutorials/concepts/logging.md
Comment (Octogonapus) [inline docs/tutorials/concepts/logging.md:6]: What good does this do?

File: docs/tutorials/walkthrough/chassisControllerBuilder.md
Comment (Octogonapus) [inline docs/tutorials/walkthrough/chassisControllerBuilder.md]: Maybe also link to this under `## Configuring odometry`?

File: docs/tutorials/walkthrough/odometry.md
Comment (Octogonapus) [inline docs/tutorials/walkthrough/odometry.md]:
```suggestion
## Configuring Odometry
```

File: docs/tutorials/walkthrough/odometry.md
Comment (Octogonapus) [inline docs/tutorials/walkthrough/odometry.md]: Can you rephrase this? It sounds awkward.

File: docs/tutorials/walkthrough/odometry.md
Comment (Octogonapus) [inline docs/tutorials/walkthrough/odometry.md]:
```suggestion
## Using Odometry
```

**Review by Octogonapus — CHANGES_REQUESTED** [review]

**Comment by theol0403** [comment]
> I kinda disagree with this. We see users in Discord not know to include the namespace, so I put it in the examples to make it clear that it uses the namespace.

It's up to you to make the final call on this, but I think that more users will read the getting started guide with the new docs. I just think it directs the user's attention to the more important parts of the example, and that it is not necessary in most user's setups.

File: docs/tutorials/concepts/logging.md
Comment (theol0403) [reply docs/tutorials/concepts/logging.md:6]: Here is how it looked before: 
image

With the line wrap the way it is, it makes it seem like the two words are part of the same link, which can be confusing.

When the user clicks on **warn**, it jumps to the anchor so that the rest of the enum is hidden:
image

In my opinion, it is more user friendly when the user clicks on **LogLevel**, because then it puts the entire enum into view:

image


Only having one link will make it so the user will be navigated to where all the modes are shown, instead of just the warn enum, which has no additional information.

File: docs/tutorials/walkthrough/odometry.md
Comment (theol0403) [reply docs/tutorials/walkthrough/odometry.md]:
```suggestion
OkapiLib supports odometry for all chassis configurations. A chassis controller that uses odometry is configured using the 
```
Better?

**Comment by theol0403** [comment]
I'm done with this PR, once we decide what to do with `using namespace okapi`.
I think v4 is ready to be released after this is merged.

In the future, a concept tutorial about the use of `auto` in the examples and how to forward declare across multiple files would be valuable.

**Review by Octogonapus — APPROVED** [review]

**Comment by Octogonapus** [comment]
> I think v4 is ready to be released after this is merged

Not quite ready because I need to setup something to show people that the v4 docs are in a new location. As far as the artifact and new docs are concerned, though, I agree they are as ready as we can get them.

**Pull request merged**
